src/adaptation/Adaptation.jl

```diff
@@ -61,15 +53,40 @@ end
 initialize!(adaptor::NaiveHMCAdaptor, n_adapts::Int) = nothing
 finalize!(aca::NaiveHMCAdaptor) = finalize!(aca.ssa)
 
-##
-## Stan's windowed adaptor.
-##
-include("stan_adaption.jl")
+include("stan_adaptor.jl")
+export NaiveHMCAdaptor, StanHMCAdaptor
+
+# User helpers for AdvancedHMC
+
+using ..AdvancedHMC: 
+    AbstractIntegrator, nom_step_size,
+    AbstractMetric, UnitEuclideanMetric, DiagEuclideanMetric, DenseEuclideanMetric
+
+StepSizeAdaptor(δ::AbstractFloat, i::AbstractIntegrator) = 
+    NesterovDualAveraging(δ, nom_step_size(i))
```

Comment:
```suggestion
    StepSizeAdaptor(δ, nom_step_size(i))
```
I feel this should be in the `integrator.jl` file instead of here. Since 

1. We don't want `Adaption` to depend on HMC code for modularity. I think we discussed this a while ago.  
2.`StepSizeAdaptor` is exported by `Adaption` anyway, so it is ok to extend it.

Reply:
I'm OK with moving them out of `adaptation.jl` to dropthis dependency but I think they should instead go to `AdvancedHMC.jl` or even a seperate file. 

My reason is that, L65 - L91 are _helper functions for adaptors_ to for easy initialization and it makes sense to group them together.

- If we put `StepSizeAdaptor` to `integrator.jl`, we should also put `MassMatrixAdaptor` to `metric.jl`. I think it's too messy and hard to find these functions.
- When I develop `integrator.jl`, I don't really care any about how adaptors would be created.

Reply:
Only `StepSizeAdaptor(δ, nom_step_size(i))` should be in `integrator.jl` since it is for Jittered leapforg integrator and depends on `nom_step_size`. Other stuff can stay as is for now.

Reply:
I don't think it makes sense. First of all, `StepSizeAdaptor(δ, nom_step_size(i))` is for all integrators but not only for jittered leapforg. `nom_step_size` defined for all integrators - for nomral leapfrog, it's nominated step size is same as actual used ones. Secondly, `MassMatrixAdaptor` also depends on metric types. It doesn't make much difference whether things are depedend on a function or just the types IMO.

Reply:
We would like to keep a one-way dependency between `Adaption` and `HMC` if possible. Before this PR, HMC depends on Adaption but not the other way in general. The `Metric` related code was part of `Adaption`'s preconditioner codes, and it might make sense to keep `Metric` part of adaption again. It is important to keep code as "simple" as possible moving forward.

Reply:
At the moment, the name `metric` is a bit abused I think. We're using `metric` to refer to both mass matrices and distributions for momentum variables (Kinect energy). However, we can probably improve this in seprate PR.

Reply:
> We would like to keep a one-way dependency ...

As I said, I agree with the dependncy point. I'm proposing moving them to `AdvancedHMC.jl` or somewhere else, instead of putting them into seperate files.

> It is important to keep code as "simple" as possible moving forward.

I belive what I proposed make the code easier to manage. In my experience of this PR, keeping these interface functions in the same place is the most easiest way, but not putting them in different places.

My preference is 1 > 2 > 3 > 4

1. Keeping them in `Adaptation.jl`
2. Keeping them in a seperate file.
3. Keeping them in `AdvancedHMC.jl` right after the include of `Adaptation`
4. Keeping them in seperate files, i.e. `MassMatrixAdaptor ` into `metric.jl` and `StepSizeAdaptor` into `leapfrog.jl`

As 1 breaks the dependency point and 2 might make the repo structure more complex, I would prefer 3 instead. 4 really makes navigating the code difficult. 

When I started with this PR and try to find this definition `NesterovDualAveraging` without search, I clicked `stepsize.jl` and then `Adaptation.jl` and didn't find it, which was quite confusing. This is the reason I think the nature place to put them are not in AHMC but in Adaptation. Later after dealing with the mass matrix adaptors, I realise that these functions are just helper functions - they are actually not for functionality of either adaptation or leapfrog/metric. They are functions only for user friendly purpose. One can use AHMC without them as well,  e.g. by doing `NesterovDualAveraging(0.8, 0.01)`. But I think those function make it clear that what are the adaptor is adapting, .e.g `StepSizeAdaptor(0.8, integrator)`.

> The Metric related code was part of Adaption's preconditioner codes, and it might make sense to keep Metric part of adaption again.

I think what we did before makes sense only because `precond.jl` contains code for adaptors but not because it contains code for metrics.

Reply:
Sure, let’s keep them in the file ‘AdvancedHMC.jl’ before we figure out a better way!